### PR TITLE
Develop

### PR DESCRIPTION
**Pull request by molleer**

### CODE REVIEW

**Comment by Epiq99** [comment]
There is no rule for code review in this repository.

**Comment by Portals** [comment]
Whyyy?

File: frontend/package.json [hunk 1]
```diff
@@ -4,40 +4,30 @@
...
-        "react-select": "^2.4.2",
         "redux": "^4.0.1",
         "redux-logger": "^3.0.6",
         "redux-mock-store": "^1.5.3",
-        "redux-thunk": "^2.3.0",
         "serve": "^11.0.2",
         "styled-components": "^4.1.3",
-        "universal-cookie": "^4.0.2",
-        "yup": "^0.27.0"
+        "universal-cookie": "^4.0.2"
```
Comment (Portals) [inline frontend/package.json]: What is this used for?

File: frontend/src/index.js [hunk 2]
```diff
@@ -2,17 +2,24 @@ import React from "react";
...
-  <SuggestITHeader renderMain = {()=>
-      <DigitProviders>
-        <Route component={App}/>
-      </DigitProviders>
-    }
-    />
-</DigitProviders>, document.getElementById("root"));
+    <DigitProviders
+        children={
+            <div>
```
Comment (Portals) [inline frontend/src/index.js]: Use Fragment instead

File: frontend/src/common/suggestitheader/suggestitheader.js [hunk 3]
```diff
@@ -1,98 +1,156 @@
...
+                                                                this.setState({
+                                                                    passTextField:
+                                                                        event.target.value,
+                                                                })
+                                                            }
+                                                            onKeyPress={event => {
+                                                                if (event.key === "Enter")
+                                                                    this.login();
+                                                            }}
+                                                            //Why not using DigitTextField, 1. No onKeyPress event, 2. No autofocus
```
Comment (Portals) [inline frontend/src/common/suggestitheader/suggestitheader.js]: Make an issue on rdc for this then 😉

File: frontend/src/use-cases/home/home.js [hunk 4]
```diff
@@ -4,32 +4,33 @@ import "./home.css";
...
-        <Prompt/>
-        {this.state.suggestionboard}
-      </div>
-    );
-  }
+    render() {
+        return (
+            <div>
+                <br />
+                <br />
```
Comment (Portals) [inline frontend/src/use-cases/home/home.js]: Don't use<br />. Use css instead.

File: frontend/src/use-cases/home/home.js [hunk 5]
```diff
@@ -4,32 +4,33 @@ import "./home.css";
...
 class Home extends Component {
-  constructor(props){
-    super(props)
-    this.state = {
-      suggestionboard: <div></div>
-    }
+    constructor(props) {
+        super(props);
+        this.state = {
+            suggestionboard: <div></div>,
```
Comment (Portals) [inline frontend/src/use-cases/home/home.js]: Render null instead

File: frontend/src/services/data.service.js [hunk 6]
```diff
@@ -1,56 +1,73 @@
...
-import Cookies from 'universal-cookie';
+import { suggestions } from "../redux/SuggestionStore";
+import axios from "axios";
+import Cookies from "universal-cookie";
 
-const cookies = new Cookies()
-const baseUrl = 'http://localhost:5000';
-const authCookieName = 'PRIT_AUTH_KEY'
+const cookies = new Cookies();
+const baseUrl = "http://suggestit.chalmers.it/api";
```
Comment (Portals) [inline frontend/src/services/data.service.js]: Use ENV?

File: frontend/src/use-cases/common/SuggestionItem/SuggestionItem.js [hunk 7]
```diff
@@ -17,71 +26,86 @@ import { connect } from 'react-redux';
...
-      </section>
-    </div>
-  </div> 
-  )
-    
-  };
+    return (
+        <div className="card">
+            <div className="innerCard">
+                <Grid container>
```
Comment (Portals) [inline frontend/src/use-cases/common/SuggestionItem/SuggestionItem.js]: Use DigitLayout.Grid 😉

File: frontend/src/use-cases/home/SuggestionBoard/SuggestionBoard.js [hunk 8]
```diff
@@ -1,84 +1,105 @@
...
-          clearButton: suggestions.getState().length > 0 ? this.ClearButton() : <DigitText.Title text="Det finns för tillfället inga förslag."/>
-        })
-      })
-      this.getData()
+class SuggestionBoardView extends Component {
+    constructor(props) {
+        super(props);
+        this.state = {
+            suggestions: [],
+            clearButton: <div></div>,
```
Comment (Portals) [inline frontend/src/use-cases/home/SuggestionBoard/SuggestionBoard.js]: Render null instead

File: frontend/package.json (hunk 1 above)
Comment (Epiq99) [reply frontend/package.json]: Creates, reads, edits and deletes cookies. Saving the token to authenticate with the backend.

File: frontend/src/common/suggestitheader/suggestitheader.js (hunk 3 above)
Comment (Epiq99) [reply frontend/src/common/suggestitheader/suggestitheader.js]: Will do

**Comment by Hoidi** [comment]
Remove .vscode folder :+1:

**Comment by Hoidi** [comment]
And a better README

File: README.md [hunk 9]
```diff
@@ -1 +1,59 @@
...
+Run the following commands to set up the development environment.  
+
+```
+    git clone git@github.com:cthit/SuggestIT.git
+    cd SuggestIT/frontend
+    yarn install
+    docker-compose up --build
+```
+The frontend is exposed to port 3000 and uses live reload.  
+The backend is exposed to port 5000 but will nor reload when the code change.
```
Comment (Hoidi) [inline README.md]: I think it's spelled not 😁

**Comment by Hoidi** [comment]
I got this when trying to build
```
Step 2/14 : RUN mkdir /usr/src/app
 ---> Running in 588e5e418122
ERROR: Service 'app' failed to build: failed to create endpoint strange_jang on network bridge: failed to add the host (veth826b55d) <=> sandbox (vethe95bbc8) pair interfaces: operation not supported
```
Not sure what it means :thinking:

File: frontend/src/use-cases/common/SuggestionItem/SuggestionItem.css [hunk 10]
```diff
@@ -25,4 +25,9 @@
 
 .footer{
```
Comment (cajoho99) [inline frontend/src/use-cases/common/SuggestionItem/SuggestionItem.css]: Why is this empty?

**Pull request merged**